privaterelay/tests/signals_tests.py

```diff
@@ -0,0 +1,80 @@
+from unittest.mock import patch
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.contrib.sessions.middleware import SessionMiddleware
+from django.http.request import HttpRequest
+from django.http.response import HttpResponse
+from django.test.client import RequestFactory
+from model_bakery import baker
+import pytest
+from waffle.models import Flag
+
+from privaterelay.signals import record_user_signed_up, send_first_email
+
+
+@pytest.fixture()
+def mock_ses_client():
+    with patch("emails.apps.EmailsConfig.ses_client") as mock_ses_client:
+        yield mock_ses_client
+
+
+@pytest.mark.django_db
+def test_record_user_signed_up_telemetry():
+    user = baker.make(User)
+    rf = RequestFactory()
+    sign_up_request = rf.get(
+        "/accounts/fxa/login/callback/?code=test&state=test&action=signin"
+    )
+
+    def get_response(_: HttpRequest):
+        return HttpResponse("200 OK")
+
+    middleware = SessionMiddleware(get_response)
+    middleware.process_request(sign_up_request)
+    record_user_signed_up(sign_up_request, user)
+
+    assert sign_up_request.session["user_created"] == True
+    assert sign_up_request.session.modified == True
+
+
+@pytest.mark.django_db
+def test_record_user_signed_up_send_first_email_requires_flag(mock_ses_client):
+    test_user_email = "testuser@test.com"
+    user = baker.make(User, email=test_user_email)
+    rf = RequestFactory()
+    sign_up_request = rf.get(
+        "/accounts/fxa/login/callback/?code=test&state=test&action=signin"
+    )
+    send_first_email(sign_up_request, user)
+
+    assert not mock_ses_client.send_email.called
+
+
+@pytest.mark.django_db
+def test_record_user_signed_up_send_first_email(mock_ses_client):
+    welcome_email_flag = Flag.objects.create(name="welcome_email", everyone=True)
```

Comment:
🤔 I'm not sure why this broke. I think using the override will fix it:

```python

from waffle.testutils import override_flag

@override_sample("welcome_email", active=True)
@pytest.mark.django_db
def test_record_user_signed_up_send_first_email(mock_ses_client):
    ...
```

Reply:
I also had to add `@pytest.mark.django_db(transaction=True)`